### DISCUSSION TITLE
More Text Font Colors and types

### DISCUSSION THREAD

**Discussion by MaMadDl** [ihhub/fheroes2#8078] [Ideas]
I'm Playing around with ICNs at the moment.
is there a need for more colors (could be done through Palettes) or more font types like **Bold** or _Italic_ ??
for font types, I think it could be done using Operations on image like Dilation

**Comment by zenseii** [comment]
Hi, @MaMadDl.

Currently we need two more fonts. One is the golden shaded one you can see in the campaign screen and highscores. It is the normal font with some gradual yellow lines and a black border.

The other is a smaller font which is used on buttons in the map editor.

However, note that we make all font changes by writing code in ui_font or agg_image. Due to copyright reasons we can't just edit/mod the original resources and share them since we do not own them.

**Reply by MaMadDl** [reply]
@zenseii we can edit output image of the font in runtime right ?

colors are based on 8bit old standard (RRRGGGBB) i added some test color in pal.cpp (it's bad shade of red ) but colors can be done easily as long as i can get the color itself right

for changing fonts since it's outputted as image (or Sprite) some standard image processing techniques ( like Dilation for bolding, skewer for italic and errosion for making the font thinner) can be done to achive this.

note that all of those are done in runtime. i'm not adding anything to AGG (for color chaning). procedures done are like how `SMALL_YELLOWFONT`  are created.

**Comment by MaMadDl** [comment]
after more tests Color coding are not standard 8-bit (RRRGGGBB)  
@zenseii do you have any thought on how the coloring works ?

**Reply by zenseii** [reply]
The expert on this subject is @ihhub. 

Having said that. For the additional font types, it would help to see drafts or examples of how it would look to make a judgement whether it isworth implementing.

**Reply by MaMadDl** [reply]
Thanks  @zenseii 
found the colors by printing values in pal2img.cpp

**Comment by MaMadDl** [comment]
Sorry For Late Reply!
I was able to spend some time on this.

Adding New font family without adding new resource is gonna be hard. but changing the existing font can done. adding new Font Colors With Different Sizes(small , normal or large) is not a problem.

next thing was a vertical color gradient that used in campaign menu. beside the font itself which is not the same other, could be done.
these are some of the examples of progress so far With Large Font and different colors

![Screenshot 2024-01-04 024946](https://github.com/ihhub/fheroes2/assets/18714088/40de28c9-35ea-4bb4-87f0-599b0e0e03f0)
![Screenshot 2024-01-04 024535](https://github.com/ihhub/fheroes2/assets/18714088/3883ea7e-6512-4678-a3ae-cfeb14f5d0c6)
![Screenshot 2024-01-04 025227](https://github.com/ihhub/fheroes2/assets/18714088/d2b66a2e-982d-445a-bab9-159fac47573a)
![Screenshot 2024-01-04 025056](https://github.com/ihhub/fheroes2/assets/18714088/516a45da-715b-4af1-9bb4-1f46f386c276)

**Reply by zenseii** [reply]
Hi, @MaMadDl. It looks like you're on the right track with how it should look. The obvious issue is how the letters don't have an expanded width but seem to have the width the normal letters have. This can be fixed by making this an entirely separate generated font ICN.

BTW I know that some of the original localizations had different design for these letters, but IMO it's best to try to recreate the English version's style of golden letters.

I'm looking forward to seeing your PR and the code for it.
